src/chain/keysinterface.rs

```diff
@@ -192,11 +198,16 @@ impl KeysInterface for KeysManager {
 	}
 
 	fn get_channel_keys(&self, _inbound: bool) -> ChannelKeys {
-		let channel_pubkey = ExtendedPubKey::from_private(&self.secp_ctx, &self. channel_master_key);
-		let mut seed = [0; 32];
-		for (arr, slice) in seed.iter_mut().zip((&channel_pubkey.public_key.serialize()[0..32]).iter()) {
-			*arr = *slice;
-		}
-		ChannelKeys::new_from_seed(&seed)
+		let mut seed = [0u8; 32 + 4 + 8 + 33];
+		rng::fill_bytes(&mut seed[..32]);
+		let now = SystemTime::now().duration_since(UNIX_EPOCH).expect("Time went backwards");
```

Comment:
Can you add a comment here something like:

"We only seriously intend to rely on the channel_master_key for true secure entropy, everything else just ensures uniqueness. We generally don't expect all clients to have non-broken RNGs here, so we also include the current time as a fallback to get uniqueness."